### DISCUSSION TITLE
v2.7.1

### DISCUSSION THREAD

**Discussion by richardcase** [kubernetes-sigs/cluster-api-provider-aws#5191] [General]
# Release notes for Cluster API Provider AWS (CAPA) v2.7.1

[Documentation](https://cluster-api-aws.sigs.k8s.io/)

# Changelog since v2.6.1

> NOTE: there is no v2.7.0 release available due to an issue encountered during the release process.

### (No, really, you MUST read this before you upgrade)

- A new community owned AWS account is now being used for publishing AMIs. The old account owner by a company is no longer accessible and the images are being automatically deleted by AWS. Support for ubuntu 18.04 & 20.04 has been dropped and ubuntu 24.04 is now the default OS if no OS is supplied. (#5133, @richardcase)
- View the projects policy on publishing AMIs [here](https://cluster-api-aws.sigs.k8s.io/topics/images/built-amis).
- There is known issue creating clusters with Kubernetes v1.30 or greater when the load balancer is an ELB (which is the default). Its recommended for new clusters that you use an NLB instead, See #5139 for further details.
 
### Support

- Add issue templates for reviewer and maintainer changes (#5079, @richardcase)
- Bump CAPI to v1.8.4 (#5061, @damdo)
- Switch `PodDisruptionBudget` in templates to policy/v1 (#5102, @AndiDog)
- Templates: align `cluster-template.yaml` and `cluster-template-machinepool.yaml`, use external cloud controller and CSI (#5136, @AndiDog)
- Update metadata file for v2.7.0 release. (#5187, @richardcase)
- Update the metadata file for v2.6.0 (#5071, @richardcase)

### Feature

- Add NonRootVolumes to AWSMachinePool launch template. (#4954, @mnitchev)
- Add ROSA Control plane registry configuration. (#5132, @serngawy)
- Change prevenet deletion label to be specific to CAPA. (#5072, @richardcase)
- Feat: add support for additionalControlPlaneIngressRules in AWSManagedControlPlane (#4783, @fad3t)
- New AMI building workflow that allows a packer vars file to be passed in. (#5152, @richardcase)
- New `AWSCluster.spec.network.NodePortIngressRuleCidrBlocks` parameter  which allows to specify IPv4 CIDR blocks for Security group rule "Node Port Services" instead of default `0.0.0.0/`. (#5147, @krasoffski)
- New workflow to publish CAPA AMIs (#5138, @richardcase)
- Support adding custom secondary VPC CIDR blocks in `AWSCluster` (#4898, @AndiDog)
- Support setting maxHealthyPercentage to configure ASG instance refresh (#5140, @fiunchinho)

### Bug or Regression

- A missing permission needed for using nlbs (#5154, @richardcase)
- Check for the LB "active" status instead of trying to resolve the DNS name to validate the LB is ready. (#5093, @r4f4)
- Fix duplicated/leaked EIP when using BYO IPv4 on Machines. (#5039, @mtulio)
- Fix kustomize issue with e2e tests. (#5167, @richardcase)
- Fix: do not attach security groups for Network Load Balancers in secret regions. (#5030, @r4f4)
- Mark `S3BucketCreated` condition ready after successful reconciliation (#5089, @AndiDog)
- Updates EKS e2e tests after recent bumping of EKS versions. (#5185, @richardcase)

### Uncategorized

- Add versionGate Acknowledgement for upgrading ROSA-HCP y-stream versions (#5166, @serngawy)
- Fix: Fargate: Standardize cluster name in role names to avoid errors on mismatch between Cluster CR and EKS cluster name (#5111, @alam0rt)
- Update Flatcar template to use external CCM. (#5099, @ader1990)

The images for this release are:

`registry.k8s.io/cluster-api-aws/cluster-api-aws-controller:v2.7.1`

Thanks to all our contributors.

<hr /><em>This discussion was created from the release <a href='https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/tag/v2.7.1'>v2.7.1</a>.</em>

**Comment by faermanj** [comment]
It seems that `cluster-template-development.yaml` is missing on this release
```
Error: failed to read "cluster-template-development.yaml" from provider's repository "infrastructure-aws": failed to download files from GitHub release v2.7.1: failed to get file "cluster-template-development.yaml" from "v2.7.1" release
```

**Reply by damdo** [reply]
I went back to the v2.1 release and looked for  `cluster-template-development.yaml`.
I don't think we ever had such file.
What component is throwing that error, is it clusterctl?
Is it configured with some special `development` flags or environment or flavour?

**Reply by faermanj** [reply]
Found this while trying to get through the quickstart.
I'll share a better review.
![image](https://github.com/user-attachments/assets/44a58f37-3c0e-4348-b149-b7823364ac06)

**Reply by richardcase** [reply]
If you click on "Other Providers" instead and run that command you should be good.

**Reply by richardcase** [reply]
So this:

```
clusterctl generate cluster capi-quickstart \
  --kubernetes-version v1.31.0 \
  --control-plane-machine-count=3 \
  --worker-machine-count=3 \
  > capi-quickstart.yaml
```

NOTE: there is no `--flavor` flag
